torch/_inductor/sizevars.py

```diff
@@ -379,59 +379,38 @@ def statically_known_power_of_2(self, expr: Expr) -> bool:
         """
         return isinstance(expr, sympy.Integer) and is_power_of_2(int(expr))
 
-    # The guard functions require you to ALREADY KNOW that a particular
-    # condition holds.  If you don't know (you want to guard on an expression
-    # being a particular value, and then get access to that value), use
-    # the evaluate functions.
+    # The expect/check functions require you to ALREADY KNOW that a particular
+    # condition holds. They are similar to expect_true in symbolic_shapes.py and
+    # torch.check but operates on sympy expressions instead of symnodes.
+    def expect_true(self, expr: Expr) -> bool:
+        if not self.statically_known_true(expr):
+            return self.shape_env.defer_runtime_assert(expr, "sizevars.expect_true")
+        return True
 
-    def guard_equals(self, left: Expr, right: Expr) -> Expr:
+    def check(self, expr: Expr) -> None:
+        assert self.expect_true(expr)
+
+    def check_equals(self, left: Expr, right: Expr) -> Expr:
+        """
+        check(sympy.Eq(left, right)) after applying inv_precomputed_replacements replacements.
+        Returns left after applying inv_precomputed_replacements.
+        """
         if isinstance(left, Expr):
             left = sympy_subs(left, self.inv_precomputed_replacements)  # type: ignore[arg-type]
         if isinstance(right, Expr):
             right = sympy_subs(right, self.inv_precomputed_replacements)  # type: ignore[arg-type]
 
-        expr = sympy.Eq(left, right)
-        static_expr = self.shape_env._maybe_evaluate_static(expr)
-
-        if static_expr is not None:
-            assert bool(static_expr)
-            return left
-
-        assert self.shape_env.defer_runtime_assert(expr, "guard_equals")
+        self.check(sympy.Eq(left, right))
         return left
 
-    def guard_leq(self, left: Expr, right: Expr) -> None:
-        return self.guard_lt(left, right + 1)
-
-    def guard_lt(self, left: Expr, right: Expr) -> None:
-        expr = sympy.Lt(left, right)
-        static_expr = self.shape_env._maybe_evaluate_static(expr)
-
-        if static_expr is not None:
-            assert bool(static_expr)
-            return
-
-        assert self.shape_env.defer_runtime_assert(expr, "guard_lt")
-
-    def guarded_order(self, seq):
-        """
-        Return the order of a sequence as a permutation of range(len(seq)) and guard on that order not changing.
-        """
-        seq = [*map(self.remove_precomputed_replacements, seq)]
-        seq = [(self.size_hint(var), orig_idx, var) for orig_idx, var in enumerate(seq)]
-        seq.sort()
-        order = [-1] * len(seq)
-        last_var = None
-        for new_index, (_, orig_index, var) in enumerate(seq):
-            order[orig_index] = new_index
-            if last_var is not None:
-                self.guard_leq(last_var, var)
-            last_var = var
-        return order
+    def check_leq(self, left: Expr, right: Expr) -> None:
```

Comment:
ditto, when/ how to use?

Reply:
not worth to add document for this it just call torch.check and I added documentation there.